server/datastore/mysql/queries_test.go

```diff
@@ -385,23 +385,16 @@ func testQueriesSave(t *testing.T, ds *Datastore) {
 	query.Query = "baz2"
 	err = ds.SaveQuery(context.Background(), query, true, true)
 	require.NoError(t, err)
+	// Ensure stats were deleted.
+	// The actual delete occurs asynchronously, so we need to wait.
+	time.Sleep(100 * time.Millisecond)
 	actual, err = ds.Query(context.Background(), query.ID)
 	require.NoError(t, err)
 	require.NotNil(t, actual)
-	// The query now comes with stats, so we need to fill them in for comparison
-	query.AggregatedStats = fleet.AggregatedStats{
-		SystemTimeP50:   ptr.Float64(0),
-		SystemTimeP95:   ptr.Float64(0),
-		UserTimeP50:     ptr.Float64(0),
-		UserTimeP95:     ptr.Float64(0),
-		TotalExecutions: ptr.Float64(1),
-	}
+	// The query now comes with stats, which are all nil
+	query.AggregatedStats = fleet.AggregatedStats{}
+	query.DiscardData = true
```

Comment:
this is already set on L355, and not seeing what would change this to to `false`

Reply:
@mostlikelee Yes, this was all my fault. I'm being too tricky with async stuff. Please re-review.